### PR TITLE
added speak plugin

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/cd95f21e7c063bb/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/cd95f21e7c063bb/output.txt

Total script time: 0.72 mins

Server is running on http://pr413.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7eded6d01f85099/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/7eded6d01f85099/output.txt

Total script time: 0.57 mins

Server is running on http://pr413.butyr.org/templates/test.html

File: templates/speak/popcorn.speak.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        if ( !target ) {
+
+          target = document.createElement( "div" );
+          target.id = options.target;
+          context.media.parentNode.appendChild( target );
+          console.log( target );
+        }
+
+        //Setup options needed for speak.js
+        if( !options.pluginPath ) { options.pluginPath = "js/plugins/speak/"; }
```
Comment (mjschranz) [inline templates/speak/popcorn.speak.js]: Can we break this onto multiple lines?

<pre>
if ( !options.pluginPath ) { 
  options.pluginPath = "js/plugins/speak/"; 
}
</pre>

File: templates/speak/popcorn.speak.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+        // SPEAK.JS by @kripken https://github.com/kripken/speak.js ---------------------------
+        try {
+          speakWorker = new Worker( options.pluginPath + 'speakWorker.js');
```
Comment (mjschranz) [inline templates/speak/popcorn.speak.js]: spacing at right bracket

File: templates/speak/popcorn.speak.js [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        } catch(e) {
```
Comment (mjschranz) [inline templates/speak/popcorn.speak.js]: need padding around the e

File: templates/speak/popcorn.speak.js [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+            var startTime = Date.now();
+            speakWorker.onmessage = function(event) {
+              if (PROFILE) console.log('speak.js: worker processing took ' + (Date.now()-startTime).toFixed(2) + ' ms');
+              handleWav(event.data);
+            };
+            speakWorker.postMessage({ text: text, args: args });
+          }
+        }
+
+      // END SPEAK.JS---------------------------------------
```
Comment (mjschranz) [inline templates/speak/popcorn.speak.js]: I hate to me mean but, can you fix all the styling in thise code you copied over from that speak.js? Just make it like everything else in regards to spacing, double quotes usage, etc.

From line 85 to here.

**Comment by jbuck** [comment]
Resolved by pull request #423

**Pull request closed without merging**